### PR TITLE
refactor: add declarative logic to reconcile PersistentVolumeClaim

### LINKED ISSUE #1321: add a declarative way  inside the codebase to reconcile PersistentVolumeClaims
Right now in our codebase, we have an imperative way to handle PVC reconciliation and this makes it hard to:
- add support for new PVCs
- reconcile the desired state without adding ad-hoc logic for every situation
- having a single point of responsibility that handles the PVC logic

We should start moving to a more declarative approach.

### PR DESCRIPTION
**Pull request by armru**
Closes #1321

### CODE REVIEW

File: api/v1/cluster_webhook.go [hunk 1]
```diff
@@ -1190,6 +1190,10 @@ func (r *Cluster) validateWalStorageChange(old *Cluster) field.ErrorList {
 		return nil
 	}
 
+	if old.Spec.WalStorage == nil && r.Spec.WalStorage != nil {
+		return nil
+	}
+
```
Comment (mnencia) [inline api/v1/cluster_webhook.go:1196]: This is caught by the previous if.

**Review by mnencia — APPROVED** [review]

**Review by gbartolini — APPROVED** [review]

**Pull request merged**